program/src/processor.rs

```diff
@@ -20,10 +19,47 @@ use {
         rent::Rent,
         slot_hashes::MAX_ENTRIES,
         system_instruction,
-        sysvar::Sysvar,
+        sysvar::{slot_hashes::SlotHashesSysvar, Sysvar},
     },
 };
 
+/// Activation status of a lookup table
+#[derive(Debug, PartialEq, Eq, Clone)]
+enum LookupTableStatus {
+    Activated,
+    Deactivating { remaining_blocks: usize },
+    Deactivated,
+}
+
+// Return the current status of the lookup table
+fn get_lookup_table_status(
+    deactivation_slot: Slot,
+    current_slot: Slot,
+) -> Result<LookupTableStatus, ProgramError> {
+    if deactivation_slot == Slot::MAX {
+        Ok(LookupTableStatus::Activated)
+    } else if deactivation_slot == current_slot {
+        Ok(LookupTableStatus::Deactivating {
+            remaining_blocks: MAX_ENTRIES.saturating_add(1),
```

Comment:
Our BPF implementation - when we went with `Clock` instead - changed this. However, the original was adding the 1.

https://github.com/anza-xyz/agave/blob/e42e7fd692a9777f0b3f25b460c2038fabbe4a2c/sdk/program/src/address_lookup_table/state.rs#L86

Come to think of it, I'm not really sure why. I wonder if it was wrong in the first place? I can check.

Reply:
Ohh I think it's because the `MAX_ENTRIES` is 512, but positions go 0 to 511. So it's supposed to be telling you that you need to wait one slot for your slot to enter the slot hashes, and then 512 more slots for it to be pushed out.

Reply:
We must have changed this when using `Clock` since it would be one less as a result of the new calculation. Maybe a small oversight with the `Clock` approach, but this seems correct to me now.

Reply:
Ohhhh I see now, that makes total sense, thanks!